### PR TITLE
Fixes LiftoverVCF for indels (again)

### LINKED ISSUE #1258: LiftoverVCF + Badly formed variant context at location chr1:899903; getEnd()
Hi,

I am using LiftoverVCF to convert hg38 to hg19 and below is the issue. Could someone help to fix this or is it a bug with picard?

```
$ picard LiftoverVcf I=114.vcf O=114.hg19.vcf CHAIN=hg38ToHg19.over.chain R=ucsc_hg19.fa REJECT=114_rejected_variants.vcf
Picked up _JAVA_OPTIONS: -Xmx64g
13:18:31.625 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/homeappl/picard/picard-tools-2.18.10/picard.jar!/com/intel/gkl/native/libgkl_compression.so
[Thu Dec 20 13:18:31 EET 2018] LiftoverVcf INPUT=114.vcf OUTPUT=114.hg19.vcf CHAIN=hg38ToHg19.over.chain REJECT=114_rejected_variants.vcf REFERENCE_SEQUENCE=ucsc_hg19.fa    WARN_ON_MISSING_CONTIG=false LOG_FAILED_INTERVALS=true WRITE_ORIGINAL_POSITION=false WRITE_ORIGINAL_ALLELES=false LIFTOVER_MIN_MATCH=1.0 ALLOW_MISSING_FIELDS_IN_HEADER=false RECOVER_SWAPPED_REF_ALT=false TAGS_TO_REVERSE=[AF] TAGS_TO_DROP=[MAX_AF] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false
[Thu Dec 20 13:18:31 EET 2018] Executing as XXXX on Linux 2.6.32-642.15.1.el6.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.10-SNAPSHOT
INFO	2018-12-20 13:18:32	LiftoverVcf	Loading up the target reference genome.
INFO	2018-12-20 13:18:56	LiftoverVcf	Lifting variants over and sorting (not yet writing the output file.)
[Thu Dec 20 13:18:57 EET 2018] picard.vcf.LiftoverVcf done. Elapsed time: 0.42 minutes.
Runtime.totalMemory()=7120355328
To get help, see http://broadinstitute.github.io/picard/index.html#GettingHelp
Exception in thread "main" htsjdk.tribble.TribbleException: Badly formed variant context at location chr1:899903; getEnd() was 899903 but this VariantContext contains an END key with value 964523
	at htsjdk.variant.variantcontext.VariantContext.validateStop(VariantContext.java:1321)
	at htsjdk.variant.variantcontext.VariantContext.validate(VariantContext.java:1294)
	at htsjdk.variant.variantcontext.VariantContext.<init>(VariantContext.java:401)
	at htsjdk.variant.variantcontext.VariantContextBuilder.make(VariantContextBuilder.java:494)
	at htsjdk.variant.variantcontext.VariantContextBuilder.make(VariantContextBuilder.java:488)
	at picard.util.LiftoverUtils.liftVariant(LiftoverUtils.java:108)
	at picard.vcf.LiftoverVcf.doWork(LiftoverVcf.java:389)
	at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:282)
	at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:103)
	at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:113)

```

### LINKED ISSUE #1280: Liftover gVCFs with indels
I think that LiftoverVcf isn't working with a gVCF and indel allele in a reference block when trying to left-align indels.

Changing this line to catch if `a.length == 0`  gets me farther (back to this issue https://github.com/broadinstitute/picard/issues/1258):
https://github.com/broadinstitute/picard/blob/master/src/main/java/picard/util/LiftoverUtils.java#L370

```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: -1
    at picard.util.LiftoverUtils.lambda$leftAlignVariant$3(LiftoverUtils.java:370)
    at java.util.stream.Collectors.lambda$groupingBy$45(Collectors.java:907)
    at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
    at java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1620)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
    at picard.util.LiftoverUtils.leftAlignVariant(LiftoverUtils.java:370)
    at picard.util.LiftoverUtils.reverseComplementVariantContext(LiftoverUtils.java:172)
    at picard.util.LiftoverUtils.liftVariant(LiftoverUtils.java:81)
    at picard.vcf.LiftoverVcf.doWork(LiftoverVcf.java:396)
    at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:295)
    at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:103)
    at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:113)
```

### PR DESCRIPTION
**Pull request by yfarjoun**
- included getStop() from rev'ed htsJDK.
- fixes (again) #1258 which was still broken due to the call to "make()" in the presence of a mismatching "END" attribute and "getEnd()" result.
- modified a test that is fixed by this commit (checked that it failing prior to the change)

fixes #1258 
fixes #1280 

Thanks to the several users who reported this is still a problem, and sorry for ignoring you for a year. @nh13, @kirannarta, and @duartemolha!

### Description

_Give your PR a **concise** yet **descriptive** title_
_Please explain the changes you made here._
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
_You can delete these instructions once you have written your PR description._

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/test/java/picard/util/LiftoverVcfTest.java [hunk 1]
```diff
@@ -1124,36 +1149,73 @@ public void testLeftAlignVariants(final VariantContext source, final ReferenceSe
...
 
-        VariantContextBuilder vcb = LiftoverUtils.liftSimpleVariantContext(source, target);
         VcfTestUtils.assertEquals(vcb == null ? null : vcb.make(), result);
     }
 
+    @Test(dataProvider = "indelNoFlipData")
+    public void testLiftOverIndels(final LiftOver liftOver, final ReferenceSequence reference, final VariantContext source, final VariantContext result) {
+
+        final Interval target = liftOver.liftOver(new Interval(source.getContig(), source.getStart(), source.getEnd()), 1);
+
```
Comment (nh13) [inline src/test/java/picard/util/LiftoverVcfTest.java]: bingo

**Comment by nh13** [comment]
@yfarjoun I am going to see if I can make a simple test case for https://github.com/broadinstitute/picard/issues/1280 and see if this fixes it.

**Comment by nh13** [comment]
@yfarjoun this fix does not solve my issue: https://github.com/broadinstitute/picard/issues/1280.  I have a single-variant test case in that PR.

**Comment by yfarjoun** [comment]
@nh13 while this PR was not intended to fix your other PR, I found the cause and I think I fixed it...care to test? (and review the PR while you're at it...)  😄

**Comment by yfarjoun** [comment]
@pshapiro4broad the problem was that the old code called "build()" on a builder that was in an improper state, the end of the variant was not at the same place where the END attribute was (for a non symbolic variant), and this didn't validate.

The changes to the tests simply adds the END attribute so that this problem is triggered, indeed the modified tests failed before I made the change.

**Comment by yfarjoun** [comment]
@nh13 you inspired me to make this work on symbolic alleles as well...and then I also figured out that I can simplify the code around the liftover that involves reverse-complementing....anyhow, no good deed will go unpunished....want to review this?

**Comment by nh13** [comment]
@yfarjoun I can review if it helps.  I won't be able to look at it until this weekend given my current workload.  Is that allright?

**Comment by yfarjoun** [comment]
Fine with me. Thanks!

**Comment by chengdai** [comment]
Any chance there has been an update to this? I'm encountering a similar error with the following genotype:

```
1       144609982       rs201830440     AATATATATATATAT A,<NON_REF>     788.73  .       DB;DP=28;ExcessHet=3.0103;MLEAC=1,1;MLEAF=0.5,0.5;RAW_MQ=68809;AC=1,1;AN=2      GT:AD:DP:GQ:PL:SB 1/2:0,19,0,0,0,0,0:19:2:824,41,194,783,0,780:0,0,9,10
```

The error message is:
```
[Fri Feb 28 19:05:05 UTC 2020] Executing as root@9823faf0b713 on Linux 4.14.154-128.181.amzn2.x86_64 amd64; OpenJDK 64-Bit Server VM 1.8.0_152-release-1056-b12; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.22.0-SNAPSHOT
INFO    2020-02-28 19:05:05     LiftoverVcf     Loading up the target reference genome.
INFO    2020-02-28 19:05:15     LiftoverVcf     Lifting variants over and sorting (not yet writing the output file.)
[Fri Feb 28 19:05:15 UTC 2020] picard.vcf.LiftoverVcf done. Elapsed time: 0.17 minutes.
Runtime.totalMemory()=6722945024
To get help, see http://broadinstitute.github.io/picard/index.html#GettingHelp
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: -1
        at picard.util.LiftoverUtils.lambda$leftAlignVariant$4(LiftoverUtils.java:379)
        at java.util.stream.Collectors.lambda$groupingBy$45(Collectors.java:907)
        at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
        at java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1620)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at picard.util.LiftoverUtils.leftAlignVariant(LiftoverUtils.java:379)
        at picard.util.LiftoverUtils.reverseComplementVariantContext(LiftoverUtils.java:178)
        at picard.util.LiftoverUtils.liftVariant(LiftoverUtils.java:76)
        at picard.vcf.LiftoverVcf.doWork(LiftoverVcf.java:426)
        at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:305)
        at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:103)
        at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:113)
```

Thanks!

**Comment by nh13** [comment]
Sorry, I'm extremely busy at work and haven't had time outside of work to look at this.  Please do not wait on my review to hold this up, as I cannot commit to doing it in the next two weeks (work + vacation)

**Pull request merged**